keystore/src/connection/platform/wasm/migration.rs

```diff
@@ -0,0 +1,198 @@
+use crate::connection::platform::wasm::version_number;
+use crate::connection::storage::{WasmEncryptedStorage, WasmStorageWrapper};
+use crate::connection::KeystoreDatabaseConnection;
+use crate::entities::{
+    E2eiAcmeCA, E2eiCrl, E2eiEnrollment, E2eiIntermediateCert, E2eiRefreshToken, Entity, EntityBase, MlsCredential,
+    MlsEncryptionKeyPair, MlsEpochEncryptionKeyPair, MlsHpkePrivateKey, MlsKeyPackage, MlsPendingMessage, MlsPskBundle,
+    MlsSignatureKeyPair, PersistedMlsGroup, PersistedMlsPendingGroup, ProteusIdentity, ProteusPrekey, ProteusSession,
+};
+use crate::{CryptoKeystoreError, CryptoKeystoreResult};
+use idb::builder::{DatabaseBuilder, IndexBuilder, ObjectStoreBuilder};
+use idb::KeyPath;
+use keystore_v_1_0_0::connection::KeystoreDatabaseConnection as KeystoreDatabaseConnectionV1_0_0;
+use keystore_v_1_0_0::entities::{
+    Entity as EntityV1_0_0, EntityFindParams as EntityFindParamsV1_0_0, MlsCredential as MlsCredentialV1_0_0,
+};
+use keystore_v_1_0_0::Connection as ConnectionV1_0_0;
+
+/// This is called from a while loop. The `from` argument represents the version the migration is performed from.
+/// The function will return the version number of the DB resulting from the migration.
+///
+/// To add a new migration, adjust the previous bottom match arm to return the current version,
+///     add a new match arm below that matches on that version, perform the migration workload
+///     and finally return `final_target`.
+pub(crate) async fn migrate(from: u32, final_target: u32, name: &str, key: &str) -> CryptoKeystoreResult<u32> {
+    const VERSION_NUMBER_V1_0_2: u32 = version_number(1, 0, 2, 0);
+    match from {
+        // The latest version that results from a migration must always map to "final_target"
+        //      to ensure convergence of the while loop this is called from.
+        0..=VERSION_NUMBER_V1_0_2 => {
+            // The version passed into this function must be the same as the one returned by this match arm.
+            // Will need to be adjusted once you add a new migration.
+            migrate_to_post_v_1_0_2(name, key, final_target).await?;
```

Comment:
A particular migration function, like `migrate_to_post_v_1_0_2` here, implicitly knows about the version it's migrating from, as well as the version it's migrating to. Given that, there should be no `final_target` in the function argument list -- the function should construct the version _internally_, and return the new db version (because the caller really shouldn't need to care about it).

Something like
```rust
   match from {
        0..=VERSION_NUMBER_V1_0_2 => migrate_from_1_0_2(name, key, ).await,
        VERSION_NUMBER_V1_1_1 => migrate_from_1_1_1(name, key, ).await,
        ...
    }
```
Note that I changed the migration function naming to mention the "from version" (which is known), instead of the "to version" (which we don't know). The rough idea here is, the migration function
- knows how to migrate _from_ a version
- it tells you which version it migrated _to_

This approach reduces the need to coordinate version numbers in several places.